deno/gateway/v10.ts

```diff
@@ -879,15 +879,17 @@ export interface GatewayGuildCreateDispatchData extends APIGuild {
 	 *
 	 * See https://discord.com/developers/docs/resources/channel#channel-object
 	 */
-	channels: APIChannel[];
+	channels: (APIChannel & { type: Exclude<ChannelType, ChannelType.DM | ChannelType.GroupDM> })[];
 	/**
 	 * Threads in the guild
 	 *
 	 * **This field is only sent within the [GUILD_CREATE](https://discord.com/developers/docs/topics/gateway-events#guild-create) event**
 	 *
 	 * See https://discord.com/developers/docs/resources/channel#channel-object
 	 */
-	threads: APIChannel[];
+	threads: (APIChannel & {
+		type: ChannelType.AnnouncementThread | ChannelType.PrivateThread | ChannelType.PublicThread;
```

Comment:
[ThreadChannelType](https://github.com/discordjs/discord-api-types/blob/main/payloads/v10/channel.ts#L208) can be used here